### PR TITLE
Fixing mirrored DICOM images and their origin

### PR DESCRIPTION
**Pull request by LouiseAmelieRB**
The method that gets the position of a slice on the stacking axis was z-specific and only worked on axial acquisition. This method was involved in both the determination of the origin of the image and the ordering of slices in the buffer.

Furthermore, the aforementioned slice ordering code is redundant with the code above it, which led to a double order inversion instead of a simple inversion of the order in which slices are stacked in the buffer, causing the images to be mirrored in case the stacking direction is the opposite of the acquisition direction.

**Before changes:**

Capture d'écran Deepin_plasmashell_20220505104345
Here on a coronal acquisition, the patient is facing the wrong way on y

Capture d'écran Deepin_plasmashell_20220505105049
Here on a sagittal acquisition, the hemispheres are mirrored, which is confirmed by opening the same data on other readers such as 3D Slicer below:
Capture d'écran Deepin_plasmashell_20220505105014

**After changes:**

Capture d'écran Deepin_plasmashell_20220505144700
Capture d'écran Deepin_plasmashell_20220505144800

**Changes:**
- Replaced `double DCMTKImageIO::GetZPositionForImage (int index)` by `double DCMTKImageIO::GetPositionOnStackingAxisForImage (int index)` to support non-axial data
- Commented the redundant change of `locSign` in `void DCMTKImageIO::ReadImageInformation()`

### CODE REVIEW

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp [hunk 1]
```diff
@@ -680,26 +687,57 @@ void DCMTKImageIO::DetermineOrientation()
...
     std::istringstream is_stream( s_position.c_str() );
-    if (!(is_stream >> junk) )
+    if (!(is_stream >> pos) )
     {
         itkWarningMacro ( << "Cannot convert string to double: " << s_position.c_str() << std::endl );
     }
-    if (!(is_stream >> junk) )
+    else
+    {
+        if (fabs(closestAxis[0]) == 1)
```
[the remark this answers is not in the document]
Comment (LouiseAmelieRB) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp:701]: done!

**Review by mathildemerle — APPROVED** [review]

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp [hunk 2]
```diff
@@ -599,8 +590,8 @@ void DCMTKImageIO::DetermineOrigin()
     int startIndex = m_FilenameToIndexMap[ m_LocationToFilenamesMap.lower_bound ( *m_LocationSet.begin() )->second ];
     int endIndex   = m_FilenameToIndexMap[ m_LocationToFilenamesMap.lower_bound ( *m_LocationSet.rbegin() )->second ];
 
-    double startZ = this->GetZPositionForImage (startIndex);
-    double endZ   = this->GetZPositionForImage (endIndex);
+    double startZ = this->GetPositionOnStackingAxisForImage (startIndex);
```
[the remark this answers is not in the document]
Comment (LouiseAmelieRB) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: Done!

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp [hunk 3]
```diff
@@ -680,26 +671,63 @@ void DCMTKImageIO::DetermineOrientation()
...
+    if (!(is_stream >> pos) )
     {
         itkWarningMacro ( << "Cannot convert string to double: " << s_position.c_str() << std::endl );
     }
-    if (!(is_stream >> junk) )
+    else
+    {
+        if (fabs(closestAxis[0]) == 1)
+        {
+            return pos;
```
[the remark this answers is not in the document]
Comment (LouiseAmelieRB) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: fixed in 2edab49

[the remark this answers is not in the document]
Comment (Florent2305) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: :+1:

File: src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Florent2305) [reply src/layers/legacy/medImageIO/itkDCMTKImageIO.cpp]: :+1:

**Review by Florent2305 — COMMENTED** [review]
Thanks for improvements

**Pull request merged**